src/settings/types.rs

```diff
@@ -44,34 +44,30 @@ impl FromStr for PythonVersion {
     }
 }
 
-#[derive(Debug, Clone, Hash)]
-pub enum FilePattern {
-    Simple(&'static str),
-    Complex(Pattern, Option<Pattern>),
-}
-
-impl FilePattern {
-    pub fn from_user(pattern: &str, project_root: Option<&PathBuf>) -> Result<Self> {
-        let path = Path::new(pattern);
-        let absolute_path = match project_root {
-            Some(project_root) => fs::normalize_path_to(path, project_root),
-            None => fs::normalize_path(path),
-        };
-
-        let absolute = Pattern::new(&absolute_path.to_string_lossy())?;
-        let basename = if pattern.contains(std::path::MAIN_SEPARATOR) {
-            None
+pub fn from_user(
```

Comment:
This name made more sense when it was part of `FilePattern`, since `FilePattern::from_user` had clear semantics. Maybe we rename to `create_glob` or `glob_from_user_pattern`?